src/client/auth/mod.rs

```diff
@@ -191,22 +228,23 @@ impl AuthMechanism {
             AuthMechanism::Plain => PLAIN_STR,
             #[cfg(feature = "aws-auth")]
             AuthMechanism::MongoDbAws => MONGODB_AWS_STR,
+            AuthMechanism::MongoDbOidc => MONGODB_OIDC_STR,
         }
     }
 
     /// Get the default authSource for a given mechanism depending on the database provided in the
     /// connection string.
     pub(crate) fn default_source<'a>(&'a self, uri_db: Option<&'a str>) -> &'a str {
-        // TODO: fill in others as they're implemented
         match self {
             AuthMechanism::ScramSha1 | AuthMechanism::ScramSha256 | AuthMechanism::MongoDbCr => {
                 uri_db.unwrap_or("admin")
             }
             AuthMechanism::MongoDbX509 => "$external",
             AuthMechanism::Plain => uri_db.unwrap_or("$external"),
+            AuthMechanism::MongoDbOidc => "$external",
             #[cfg(feature = "aws-auth")]
             AuthMechanism::MongoDbAws => "$external",
-            _ => "",
```

Comment:
The catchall here caused me a bit of lost time debugging an unhelpful server `invalid database name ""` error; since the only thing it caught was `Gssapi`, updated to that to avoid repeating the problem if we add more auth methods.